polkadot/runtime/parachains/src/shared.rs

```diff
@@ -43,16 +45,28 @@ pub(crate) const SESSION_DELAY: SessionIndex = 2;
 #[cfg(test)]
 mod tests;
 
-/// Information about past relay-parents.
+pub mod migration;
+
+/// Information about a relay parent.
+#[derive(Encode, Decode, Default, TypeInfo, Debug)]
+pub struct RelayParentInfo<Hash> {
+	// Relay parent hash
+	pub relay_parent: Hash,
+	// The state root at this block
+	pub state_root: Hash,
+	// Claim queue snapshot, optimized for accessing the assignments by `ParaId`.
+	// For each para we store the cores assigned per depth.
+	pub claim_queue: BTreeMap<Id, BTreeMap<u8, BTreeSet<CoreIndex>>>,
+}
+
+/// Keeps tracks of information about all viable relay parents.
 #[derive(Encode, Decode, Default, TypeInfo)]
 pub struct AllowedRelayParentsTracker<Hash, BlockNumber> {
-	// The past relay parents, paired with state roots, that are viable to build upon.
+	// Information about past relay parents that are viable to build upon.
 	//
 	// They are in ascending chronologic order, so the newest relay parents are at
 	// the back of the deque.
-	//
-	// (relay_parent, state_root)
-	buffer: VecDeque<(Hash, Hash)>,
+	buffer: VecDeque<RelayParentInfo<Hash>>,
```

Comment:
since we're making breaking changes to a storage that is used by a runtime API, we should make sure to workaround the recently experienced issue: #5738 by adjusting it https://github.com/paritytech/polkadot-sdk/blob/d31bb8ac990286f4d4caa759ea960bb9866fc2b0/polkadot/runtime/parachains/src/runtime_api_impl/v10.rs#L454

Reply:
🤦🏼‍♂️  good catch, I missed this, was thinking it is only used internally and not exposed via API.

Reply:
don't see which commit fixed that 👀

Reply:
fixed now.